src/Microsoft.HttpRepl/ApiConnection.cs

```diff
@@ -118,19 +137,19 @@ public void SetupApiDefinition(HttpState programState)
             }
         }
 
-        public async Task SetupHttpState(HttpState httpState, bool performAutoDetect, CancellationToken cancellationToken)
+        public async Task SetupHttpState(HttpState httpState, bool performAutoDetect, VerbosityLogger logger, CancellationToken cancellationToken)
```

Comment:
Would you ever pass different loggers to a single instance of this?  I wonder if it would be nicer to take as a construction parameter, and then you don't have to pass this through every call.